apps/website/src/components/Badges.tsx

```diff
@@ -0,0 +1,52 @@
+import type { ApiDocumentedItem } from '@microsoft/api-extractor-model';
+import { ApiAbstractMixin, ApiProtectedMixin, ApiReadonlyMixin, ApiStaticMixin } from '@microsoft/api-extractor-model';
+import type { ReactNode } from 'react';
+
+const mappedBadgeColors = {
+	danger: 'bg-red-500',
+	warning: 'bg-yellow-500',
+	primary: 'bg-blurple',
+} as const;
+
+export type BadgeColor = keyof typeof mappedBadgeColors;
+
+export interface BadgeProps {
+	/**
+	 * The content of the badge.
+	 */
+	children?: ReactNode | undefined;
+	/**
+	 * The color of the badge.
+	 *
+	 * @defaultValue "primary"
+	 */
+	color?: BadgeColor;
+}
+
+export function Badge({ children, color = 'primary' }: BadgeProps) {
```

Comment:
Use `PropsWithChildren` from the react typings, passing additional props as a generic.